packages/data/src/index.js

```diff
@@ -367,8 +377,8 @@ export const withDispatch = ( mapDispatchToProps ) => createHigherOrderComponent
 					this.setProxyProps( props );
 				}
 
-				componentWillUpdate( nextProps ) {
-					this.setProxyProps( nextProps );
+				componentDidUpdate() {
```

Comment:
I don't think this conversion is correct: `componentWillUpdate` runs before `render`, and `setProxyProps` is only assigning an instance variable, so on its own it won't incur another render. Therefore, by changing to `componentDidUpdate` we risk the new proxy props not being applied correctly (or at least until the next render).

Reply:
Yes, I was aware of this difference but in my testing it's working fine. What alternative do you propose?